### DISCUSSION TITLE
Is the intent to always serialize Lexical state to the database?

### DISCUSSION THREAD

**Discussion by sherrellbc** [facebook/lexical#5512] [General]
I have a simple application that instantiates a `PlainTextPlugin` that has some simple markdown processing. The users can save the contents and then later edit them. I can easily get the text-based contents of the editor (and save to my database) but it is not simple (impossible?) to create a new Lexical editor and _restore_ this text.

```
function initializeEditor(editor, initialEditorState) {
  if (initialEditorState === null) {
    return;
  } else if (initialEditorState === undefined) {
    editor.update(() => {
      const root = lexical.$getRoot();
      if (root.isEmpty()) {
        const paragraph = lexical.$createParagraphNode();
        root.append(paragraph);
        const activeElement = CAN_USE_DOM ? document.activeElement : null;
        if (lexical.$getSelection() !== null || activeElement !== null && activeElement === editor.getRootElement()) {
          paragraph.select();
        }
      }
    }, HISTORY_MERGE_OPTIONS);
  } else if (initialEditorState !== null) {
    switch (typeof initialEditorState) {
      case 'string':
        {
          const parsedEditorState = editor.parseEditorState(initialEditorState);
          editor.setEditorState(parsedEditorState, HISTORY_MERGE_OPTIONS);
          break;
        }
      case 'object':
        {
          editor.setEditorState(initialEditorState, HISTORY_MERGE_OPTIONS);
          break;
        }
      case 'function':
        {
          editor.update(() => {
            const root = lexical.$getRoot();
            if (root.isEmpty()) {
              initialEditorState(editor);
            }
          }, HISTORY_MERGE_OPTIONS);
          break;
        }
    }
  }
}
```

I see no way to restore Lexical editor state from string contents - only from a serialized JSON state or from an actual `EditorState` object. This is unexpected, since by requiring the Lexical state to be _the_ source of state means the the content stored in the database must necessarily be tightly coupled with Lexical. This makes it difficult to e.g. just render the text content or switch frontend editors.

I understand that it does not make sense for something like _RichTextPlugin_ since there is additional state for formatting and such that will be lost, but there really should be some way to say "set the editor's text content to this string", right?

**Comment by CanRau** [comment]
Hope you've figured it out by now, commenting just in case and potentially for others

It depends, in our case we store a plaintext version and the full json editor state in the db, especially cause we're working with rich text as you mentioned

Still I have this to restore from plaintext in certain cases or where I later changed a `<textarea>` into a Lexical editor

```ts
const initialConfig: InitialConfigType = {
  editorState(editor) {
    editor.update(() => {
      const root = $getRoot();
      const splitted = plaintext.split(/\r?\n|\r|\n/g);
      splitted.forEach((v) => {
        const paragraphNode = $createFederParagraphNode();
        const textNode = $createFederTextNode(v);
        paragraphNode.append(textNode);
        root.append(paragraphNode);
      });
    });
  },
};

```
This ensures that paragraphs are properly restored and is sufficient in our case so far, you could of course tweak it to your needs.

Hope this helps ❤️
